lib/vsc/filesystem/quota/itools.py

```diff
@@ -0,0 +1,184 @@
+    # Note: other settings, e.g., ofr each cluster will be obtained from the configuration file
+    CLI_OPTIONS = {
+        'nagios-check-interval-threshold': NAGIOS_CHECK_INTERVAL_THRESHOLD,
+        'location': ('path to store the gzipped files', None, 'store', INODE_LOG_ZIP_PATH),
+        'backend': ('Storage backend', None, 'store', 'gpfs'),
+        'host_institute': ('Name of the institute where this script is being run', str, 'store', GENT),
+        'mailconfig': ("Full configuration for the mail sender", None, "store", None),
+    }
+
+    def mail_admins(self, critical_filesets, dry_run=True, host_institute=GENT):
+        """Send email to the HPC admin about the inodes running out soonish."""
+        mail = VscMail(mail_config=self.options.mailconfig)
+
+        message = CRITICAL_INODE_COUNT_MESSAGE
+        fileset_info = []
+        for (fs_name, fs_info) in critical_filesets.items():
+            for (fileset_name, inode_info) in fs_info.items():
+                fileset_info.append("%s - %s: used %d (%d%%) of max %d [allocated: %d]" %
+                                    (fs_name,
+                                    fileset_name,
+                                    inode_info.used,
+                                    int(inode_info.used * 100 / inode_info.maxinodes),
+                                    inode_info.maxinodes,
+                                    inode_info.allocated))
+
+        message = message % ({'fileset_info': "\n".join(fileset_info)})
+
+        if dry_run:
+            logging.info("Would have sent this message: %s", message)
+        else:
+            mail.sendTextMail(mail_to=INSTITUTE_ADMIN_EMAIL[host_institute],
+                            mail_from=INSTITUTE_ADMIN_EMAIL[host_institute],
+                            reply_to=INSTITUTE_ADMIN_EMAIL[host_institute],
+                            mail_subject="Inode space(s) running out on %s" % (socket.gethostname()),
+                            message=message)
+
+
+    def do(self, dry_run):
+        """
+        Get the inode info
+        """
+        stats = {}
+
+        backend = self.options.backend
+        if backend == 'gpfs':
+            storage_backend = GpfsOperations()
+        elif backend == 'lustre':
+            storage_backend = LustreOperations()
+        else:
+            logging.error("Backend %s not supported", backend)
+            raise
+
+        filesets = storage_backend.list_filesets()
+        quota = storage_backend.list_quota()
+
+        if not os.path.exists(self.options.location):
+            os.makedirs(self.options.location, 0o755)
+
+        critical_filesets = dict()
+
+        for filesystem in filesets:
+            stats["%s_inodes_log_critical" % (filesystem,)] = INODE_STORE_LOG_CRITICAL
+            try:
+                filename = "%s_inodes_%s_%s.gz" % (backend, time.strftime("%Y%m%d-%H:%M"), filesystem)
+                path = os.path.join(self.options.location, filename)
```

Comment:
@itkovian as this package is py3_only now, I suggest replacing os.path statements with pathlib which is more of a pythonic way of doing things. Only a suggestion though.

Reply:
maybe in a followup PR :)